### PR TITLE
[ELY-2683] Refactor OAuth2SaslClientV11Test to use common method

### PR DESCRIPTION
**Pull request by ericrleung**
https://issues.redhat.com/browse/ELY-2683

### CODE REVIEW

File: tests/base/src/test/java/org/wildfly/security/sasl/oauth2/OAuth2SaslClientV11Test.java [hunk 1]
```diff
@@ -497,10 +497,7 @@ public void testWithResourceOwnerCredentialsInCredentialStoreUsingConfiguration(
         assertTrue(saslClient.isComplete());
     }
 
-    @Test
-    public void failedResourceOwnerCredentialsUsingConfiguration() throws Exception {
-        SaslClient saslClient = createSaslClientFromConfiguration(URI.create("protocol://test9.org"));
-
+    public void failedCredentialsUsingConfiguration(SaslClient saslClient) throws Exception {
```
Comment (skyllarr) [inline tests/base/src/test/java/org/wildfly/security/sasl/oauth2/OAuth2SaslClientV11Test.java]: @ericrleung Just a minor, you can pass "String" serverURI to this method and create a SaslClient there

**Review by skyllarr — CHANGES_REQUESTED** [review]
SaslClient can be created in the new method

File: tests/base/src/test/java/org/wildfly/security/sasl/oauth2/OAuth2SaslClientV11Test.java (hunk 1 above)
Comment (skyllarr) [reply tests/base/src/test/java/org/wildfly/security/sasl/oauth2/OAuth2SaslClientV11Test.java]: @ericrleung Just a minor, you can then rename this method so its name is more generic

Comment (ericrleung) [reply tests/base/src/test/java/org/wildfly/security/sasl/oauth2/OAuth2SaslClientV11Test.java]: @Skyllarr What should I change the name to?

Comment (fjuma) [reply tests/base/src/test/java/org/wildfly/security/sasl/oauth2/OAuth2SaslClientV11Test.java]: Maybe testInvalidCredentials or testInvalidCredentialsUsingConfiguration?

@Skyllarr Did you have something else in mind?

Comment (skyllarr) [reply tests/base/src/test/java/org/wildfly/security/sasl/oauth2/OAuth2SaslClientV11Test.java]: yes those sound good

**Comment by PrarthonaPaul** [comment]
Hi @ericrleung 
It looks like there are some comments on this PR that still need to be addressed. 
If you have questions about any of the comments, please feel free to let us know! 
Thank you.

**Review by fjuma — APPROVED** [review]
Thanks @ericrleung!

**Review by skyllarr — APPROVED** [review]
Thanks @ericrleung

**Pull request merged**